### ISSUE TITLE
any plans for open source?

### ISSUE THREAD

**Issue by MF-Debug** [8bitPit/Niagara-Issues#180]
do you wish or have any plans to make this app open source in future? many people prefer open source apps from f-droid including myself. so that would be a great decision!

**Comment by 8bitPit** [comment]
Sorry, I don’t have plans for now, a project should only be open sourced if you’re 100% certain that this would be the best option, and that’s not what I am yet. I know, open sourcing it would be a more transparent approach, however I fear of losing uniqueness and control. The problem is, you cannot go back.

**Comment by jakcharvat** [comment]
Perhaps having a very basic vanilla version of the launcher open sourced? Say the fist public beta build? The rest could be private. Just a though. Slightly like how Google has Chromium and Chrome, except you wouldn't need to maintain the open source version

**Comment by nonnymoose** [comment]
@8bitPit I highly recommend open-sourcing it if you would want contributions from the community. Instead of having nearly so many feature requests, you would get a small number of feature contributions from people who are willing to write in the feature they want.

**Comment by 8bitPit** [comment]
@jakcharvat  interesting idea, will definitely keep that in mind!
And yes, @nonnymoose you’re right, hope you can both understand that I’ll take some time to decide if I open source / partly open source it.

Sounds also good for a Patreon stretch goal :P 
(if I ever use that donation service)

**Comment by OverRidingSky** [comment]
I'd personally use an open-source build as a guide. I've been googling here and there to find out how to use basic features other homescreens / launchers take advantage of, but looking at other's source is a lot easier then going stir crazy finding out what the person means to say.

**Comment by cauerego** [comment]
ideas aren't unique, implementations are. nobody is ever going to be able to "steal" the way we continually build our ideas. plus, rest assured, no idea is so original that aren't many other similar ones already.

oss always make more sense, no matter how you look at it.

and you can go back.

in practice, though, there are many ways to open it up. you can simply release the code without putting any effort in making it readable to others. or you can go the opposite and much costly easy of making it easy to digest. and anywhere in between.

opening up just the core, for instance, is about in the middle.

and in the grand scheme, sharing our ideas openly is the best way to grow them. because it grows both from people willing to help and from the ones looking to take any kind of unfair advantage.

bottom line, even if at this point there's nobody serious about contributing and it may look like there's no benefit in going through any effort of going free as in free hugs, truth is the longer you take to do it, the harder it could be for contributors to believe they'd be joining a good cause.

just the 2 cents of a poor man.

**Comment by 8bitPit** [comment]
Thanks for joining the discussion @cauerego. Regarding your points:

> you can go back.
You can technically go back, but your code, or copies / forks of your project remain online. And that cannot be undone, if that’s really the best way for the project. 

> in the grand scheme, sharing our ideas openly is the best way to grow them. because it grows both from people willing to help and from the ones looking to take any kind of unfair advantage.
It’s not that I don’t consider other people’s ideas just because Niagara is closed source. It’s right, that it’s harder to contribute written lines of code when you don’t know the source. However, I believe Niagara grows well already and open sourcing it doesn’t speed up the process that much, since Niagara is not based on Launcher 3 (most open source launchers are) and it’s harder to as a programmer to orient themselves there. It’s difficult to write clean and well performing code when you’re not aware of your “surroundings”. Furthermore, it isn’t a good idea just to open source it; for proper implementations I would have to introduce guidelines, arrange meetings (so the launcher remains consistent & minimal), explain a lot of my code and so on. Imo that doesn’t pay off yet - as I said Niagara already flourishes well with the help of the community, even without open sourcing it. 

> oss always make more sense, no matter how you look at it.
I believe that’s not always the case. For example, it’s definitely harder to run a business with open source projects. I know, I am not part of a company, but I would like to offer additional features with a pro version. 
I appreciate your willingness to help and I hope you understand, why I am not open sourcing Niagara yet.

**Comment by zamber** [comment]
> ideas aren't unique, implementations are. nobody is ever going to be able to "steal" the way we continually build our ideas. plus, rest assured, no idea is so original that aren't many other similar ones already.

The fact that ideas are not unique does not justify giving an implementation out for free.

Open sourcing Niagara would make sense if Peter would want to abandon the project and give it back to the community, then GPL3 and YOLO. Everyone lives happily ever after.

If he would want to continue development then GPL can (and probably will) bite him back because it's cutting all possibility of any sensible profit. Prove me wrong with an example of a successful, profitable GPL-licensed Android app.

If he'll get a more permissive license then there's nothing stopping, say Samsung from taking the codebase and incorporating it into their crapware package. That's why you usually see BSD in libraries, not finished apps. Sell support not the product is a proven business model in OSS.

Any kind of utopian principles are dangerous on the free market unless there's a chance for disruption. Think Linux, think LibreOffice, think Blender. For a mere Android launcher it makes more sense to stay closed-source because of this. Again, point us (or actually Peter) to an open-source Android app with global reach and/or a business model that would make it sustainable.

There's also real risk that <insert crapware app vendor> will take it, add a pile of tracking and features on top and kill Niagara in the process. It wouldn't be the 1st time a fork has to take over an OSS project. Adblock Plus anyone?

Regarding feature contributions I doubt Peter would want to get the responsibility package of a BDFL. It's more likely that the project would die a death of a 1000 PRs. I would be also skeptical about providing any API or core implementation for PRs due to the aforementioned issue and let's top it up with all of the security implications that may follow (now not an issue but it will be if Peter decides to add, say, contacts to search).

> and in the grand scheme, sharing our ideas openly is the best way to grow them. because it grows both from people willing to help and from the ones looking to take any kind of unfair advantage.

In the grand scheme someone will re-implement this launcher. Possibly Nova or AL as they like to give a crapton of feature combinations. I also don't see them open-sourcing any time soon, so what's the point in inviting

> the ones looking to take any kind of unfair advantage.

?

> truth is the longer you take to do it, the harder it could be for contributors to believe they'd be joining a good cause.

Technically that's not a truth, it's an authoritarian statement with a dose of unsureness ("could be"...). I see that type of argumentation coming out from people talking about politics, religion and "open-source in name of principles". RMS at least follows up with a whole belief system on top of OSS.

Personal opinion corner: I see this discussion as pretty toxic for the OSS community as a whole. There are no proof-supported selling points here for a small project to go open-source. Persuasion based on, essentially, beliefs and optimistic projections is cheap. The fact that someone does not believe in intellectual property does not justify advocating for abandoning all the protections that the aforementioned IP law offers in name of essentially an ideology.

**Comment by ZVNexus** [comment]
> If he would want to continue development then GPL can (and probably will) bite him back because it's cutting all possibility of any sensible profit. Prove me wrong with an example of a successful, profitable GPL-licensed Android app.

For one, Phonograph Pro.

You can open source the entire project but only build small portions of it. Then you can sell the complete package for 2.49 or something like that. 
People who know how to build it themselves can build the entire thing for free if they like, its not like the developer would be losing much money at all. The amount of people who would be downloading this launcher that actually know how to build it themselves would be extremely slim. 

However, Phonograph did have forks and were blatantly copy pasted onto the Play Store. Vinyl Music Player is one such example. The forks generally will never overtake the actual main project, Vinyl has much less downloads than Phonograph. Its a possibility though.

**Comment by deadcombo** [comment]
This is a complex issue, but I feel that this is completely up to Peter to decide. I generally prefer to go with FOSS apps because of trust, mostly in regards to personal data.

Going through the issues here, and by the way the app is built, this seems to be a trustworthy project (well, I still disabled the tracking and firewalled it, but I'm also paranoid). I'm using it, I like it, this is the first time I've switched launcher since Nova came out (which is another app that's not open source and that I trusted much the same way I trust Niagara).

I don't think it's impossible to trust closed source software, but it takes a lot of work from the developer to build that trust. Honestly though, when it comes the time for you to profit from here, I don't think this will matter much. You have a very good product here, one that's gaining traction here. As I see it, in the Android ecosystem most people who'll pay for an app don't care about the source. And you don't seem to be in need of much help to continue developing, so I guess there's not much of an upside for you to open source. 


On a personal note, I'd very much like to see a version of Niagara that didn't come with tracking (especially not Google's) and that was GSF-independent. This I would value just an much as it being open-source, and would gladly pay for.

Thanks for reading, and for creating a marvelous launcher.

**Comment by paulie-g** [comment]
It's not about trust for me personally - you accept a large amount of the code you're going to be running on Android will be non-free and proprietary. Heck, I can't even trust the manufacturer not to stick Qualcomm's rootkit on my device.

My problem is this - I like the launcher. It is not, however, currently usable. I'd like to be able to fix that and submit PRs. I can't, because it's closed. Imo, the decision is between having a perennially unfinished closed project or an open project that has a chance of getting up to some reasonable standard of features and plain working properly. There's a config toggle that doesn't work. This would've taken me 15 minutes to fix (10 of them figuring out the build), and Peter would have had a PR ready to merge.

Just my 2 deutschmarks, obviously Peter's call.

**Comment by 8bitPit** [comment]
Hey, it's me again after quite some time. We decided to not make the launcher open source. Nevertheless thanks for being part of the discussion, I found it pretty interesting. Sorry that this isn't really a response to your arguments, but I fear this discussion would go on forever. Personally, I wouldn't say that FOSS isn't better or worse than closed source projects, they are different approaches with different advantages and disadvantages. If you prefer open source launchers though, I recommend Lawnchair Launcher or KISS Launcher if you prefer a list-based approach. Again, thanks for sharing your thoughts.

**Comment by cauerego** [comment]
thanks so much for recommending kiss, peter!

just tried it briefly, and it looks amazing!!

you certainly mentioned it before, but now it's the first time i see a good
reason to try another launcher...

OSS is a crucially big deal to me.

good luck there and cheers!

**Comment by andys8** [comment]
From my experience all issues I currently have and had with the launcher were minor things, that could be easily fixed or adapted. The version updates containing fixes each lead to a new issues. Opensource would have allowed fixes by the community or installing a fix instead of changing the launcher (rollback wouldn't have helped).

Of course the decision to have a project open or closed source is 100 % in the hands of the author, and fine either way. In the future I'll try to prefer and support open source projects with patreon, GitHub sponsors or pro version over closed source projects.

**Comment by 8bitPit** [comment]
Agreed, that's one of the advantages of Open source projects

**Comment by gary-host-laptop** [comment]
![uCaOsxArH5](https://github.com/8bitPit/Niagara-Issues/assets/63723030/177d6ec9-30a3-4c5b-88fb-87240fc64b04)

**Comment by NicoWeio** [comment]
…But really: With how much Niagara has evolved since the creation of this issue, I feel like the release of at least a "base" version should be less daunting.
(I hope you don't mind the ping since it's been 3 years since this was last discussed.)

**Comment by abusager13** [comment]
I still believe that this app will be open sourced sometime in the future
